### PR TITLE
Front end login

### PR DESCRIPTION
**Pull request by Valleyleaf**
Added Mutations and Queries. Added Profile screen. Need Course componenet before finalizing, currently has a lot of placeholder elements.

### CODE REVIEW

File: client/src/components/mainComponents/Login.jsx [hunk 1]
```diff
@@ -42,20 +47,23 @@ const Login = () => {
...
               type='text'
               name='name'
               value={formInput.name}
               onChange={handleChange}
-              placeholder="Enter your username"
+              placeholder="Enter your email"
               onBlur={handleBlurName}
             />
-            <h4 id='warningTextNameID' className='wt warningTextForm hidden-element'>*Please insert a userame</h4>
+            <h4 id='warningTextNameID' className='wt warningTextForm hidden-element'>*Please insert a email</h4>
```
Comment (bhansi) [inline client/src/components/mainComponents/Login.jsx]: Grammar: Please insert *an* email

File: client/src/main.jsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import Courses from './components/mainComponents/Courses.jsx'
 import Login from './components/mainComponents/Login.jsx'
 import LearnMore from './components/mainComponents/LearnMore.jsx'
 import Signup from './components/UI/Signup.jsx'
+import Profile from './components/UI/ProfilePage.jsx'
```
Comment (bhansi) [inline client/src/main.jsx]: Should this be 'import ProfilePage' instead of 'import Profile'?

File: client/src/utils/mutations.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  mutation addProfile($username: String!, $email: String!, $password: String!) {
+    addProfile(name: $username, email: $email, password: $password) {
+      token
+      profile {
+        _id
+        name
+      }
+    }
+  }
+`;
```
Comment (bhansi) [inline client/src/utils/mutations.js]: The backend is doing firstName and lastName instead of username, let's discuss which one we would like to go with.

**Review by bhansi — COMMENTED** [review]
Need to discuss parameter names for queries and mutations to match them up with backend.

File: client/src/components/mainComponents/Login.jsx (hunk 1 above)
Comment (Valleyleaf) [reply client/src/components/mainComponents/Login.jsx]: Fixing

File: client/src/main.jsx (hunk 2 above)
Comment (Valleyleaf) [reply client/src/main.jsx]: I'll change it to ProfilePage for considency. It doesn't really impact anything in terms of functionality

File: client/src/utils/mutations.js (hunk 3 above)
Comment (Valleyleaf) [reply client/src/utils/mutations.js]: Let's stay with what you guys have going on the backend.

**Review by MagicianJoshua — APPROVED** [review]
approving

**Pull request merged**